games/static/games/proto.js

```diff
@@ -1,14 +1,48 @@
+class Game{
+    constructor() {
+
+    }
+
+}
+
+class Setup{
+    constructor(alienLocations, gameSpeed, movementPattern) {
+        this.alienLocations = alienLocations
+        this.gameSpeed = gameSpeed
+        this.movementPattern = movementPattern
+    }
+    getAlienLocations() {
+        return this.alienLocations
+    }
+    getGameSpeed() {
+        return this.gameSpeed
+    }
+    getMovementPattern() {
+        return this.movementPattern
+    }
+}
+
 class Square{
-    constructor(indexOnBoard, div) {
+    constructor(indexOnBoard, div, lineWidth) {
         this.index = indexOnBoard
         this.div = div
+        this.lineWidth = lineWidth
     }
     setCharacter(character) {
         this.character = character
         let cssClass = character.getCssClass()
         this.div.classList.add(cssClass)
     }
-    isTaken() {}    
+    hasAlien() {
+        return (this.classList.contains('invader') || this.classList.contains('intruder'))
+    } 
+    hasDefender() {
+        return (this.classList.contains('defender'))
+    }
+    topOfBoard() {
+        return (this.indexOnBoard <= this.lineWidth)
+    }
+
 }
 
 class Defender{
```

Comment:
does Defeder really needs to know its index ? i mean its standing on a square that holds the index